static/js/util.ts

```diff
@@ -208,28 +218,28 @@ export function sorted_ids(ids) {
     return id_list;
 }
 
-export function set_match_data(target, source) {
+export function set_match_data(target: Message, source: MatchedMessage): void {
```

Comment:
It looks like `source` may be either `MatchedMessage` or `Message`; we could use the common supertype `Partial<MatchedMessage>`.

https://github.com/zulip/zulip/blob/7fd8d77ce0fe66008e4656d8ef6ef92f35bf0904/static/js/message_events.js#L60

https://github.com/zulip/zulip/blob/7fd8d77ce0fe66008e4656d8ef6ef92f35bf0904/static/js/message_helper.js#L21